contextualized/modules.py

```diff
@@ -45,8 +45,12 @@ class MLP(nn.Module):
     """
     def __init__(self, input_dim, output_dim, width, layers, activation=nn.ReLU, link_fn=identity_link):
         super(MLP, self).__init__()
-        hidden_layers = lambda: [layer for _ in range(0, layers - 2) for layer in (nn.Linear(width, width), activation())]
-        mlp_layers = [nn.Linear(input_dim, width), activation()] + hidden_layers() + [nn.Linear(width, output_dim)]
+        if layers > 0:
+            mlp_layers = [nn.Linear(input_dim, width), activation()] + \
+                    [layer for _ in range(0, layers - 1) for layer in (nn.Linear(width, width), activation())] + \
```

Comment:
Should the order here be reversed to this?

`[layer for layer in (nn.Linear(width, width), activation()) for _ in range(0, layers - 1)]`

Reply:
Probably should be rewritten for readability without a nested iterator, either way.

Reply:
Fixed, marking resolved